src/utils/plot_results.jl

```diff
@@ -138,56 +132,28 @@ function get_bar_gen_data(res::OperationsProblemResults)
 end
 
 """
-    sort_data(results::OperationsProblemResults)
+    sort_data!(results::OperationsProblemResults)
 
 This function takes in struct OperationsProblemResults,
 sorts the generators in each variable, and outputs the sorted
 results. The generic function sorts the generators alphabetically.
 
-kwargs: 'Variables' to choose which variables to be sorted.
+# Arguments
+-`results::OperationsProblemResults`: the results of the simulation
 
-each variable has a kwarg, so that a specific order can
-be generated, such that when plotted, the first generator is on the bottom.
-if a list of generator names has fewer generators than the variable, only the
-generators on the list will be outputted.
+# Accepted Key Words
+-`Variables` to choose which variables to be sorted.
 
 #Examples
-
-example 1:
-sorted_results = sort_data(res)
-
->sorted_results.variables[P_RenewableDispatch] will be in the order
-    [:WindBusA :WindBusB :WindBusC] (alphabetical)
-
-example 2:
-my_order = [:WindBusC :WindBusB :WindBusA]
-sorted_results = sort_data(res; P_RenewableDispatch = my_order)
-
->sorted_results.variables[P_RenewableDispatch] will be in the order
-    [:WindBusC :WindBusB :WindBusA] (my_order)
-
-example 3:
-my_order = [:WindBusC :WindBusA]
-sorted_results = sort_data(res; P_RenewableDispatch = my_order)
-
->sorted_results.variables[P_RenewableDispatch] will be in the order
-    [:WindBusC :WindBusA] (my_order)
-
-example 4:
-my_variable_order = [:P_ThermalStandard :ON_ThermalStandard]
-sorted_results = sort_data(res; Variables = my_variable_order)
-
->sorted_results.variables
-    Dict{Symbol,DataFrames.DataFrame} with 2 entries:
-    :P_ThermalStandard => 24×5 DataFrames.DataFrame…
-    :ON_ThermalStandard => 24×5 DataFrames.DataFrame…
-
-* note that only the generators included in 'my_order' will be in the
+```julia
+sort_data!(results)
+```
+***Note:*** only the generators included in 'my_order' will be in the
 results, and consequently, only these will be plotted. This can be a nice
 feature for variables with more than 5 generators.
 
 """
-function sort_data(res::OperationsProblemResults; kwargs...)
+function sort_data!(res::OperationsProblemResults; kwargs...)
```

Comment:
Why did you add the '!'?  It does look like res is being indirectly modified, but then it returns a new instance.  Semantically, when I read the function signature I would assume that this will sort res in place and not return anything.